docs/design/enqueue-action.md

```diff
@@ -0,0 +1,59 @@
+behavioural characteristics how scheduler make scheduling decisions.
+
+## Enqueue Action
+
+Enqueue is one of the actions in Volcano scheduler.  Enqueue action comes into play when
+a new job is created, this action will filter qualified jobs into the queue to be scheduled.
+
+When the minimum number of resource requests under a job can not be met, even if the scheduling action is 
+performed for pod under a Job, pod will not be schedule because the "Gang" constraint is not reached. 
+state of job only allowed change from `Pending` to `Inqueue` if the minimum resource size of the job is met.
+
+Enqueue action is the preparatory stage in the scheduling process. it can prevent a large number of 
+unscheduled pods in the cluster and improve the performance of the scheduler in the high-load scenarios where 
+the cluster resources may be insufficient. Therefore, the Enqueue action is an essential action for the scheduler configuration.
+
+In Enqueue Action, there are multiple plugin functions that are getting used like
+
+1. QueueOrderFn(Plugin: Priority, DRF, Proportion),
+2. JobOrderFn(Plugin: Priority, DRF, Gang),
+3. JobEnqueueable(Plugin: OverCommit, Proportion, ResourceQuota, SLA),
+
+### 1. QueueOrderFn:
+#### Priority:
+Compares queuePriority set in Spec(using PriorityClass) and returns the decision of comparison between two priorities.
+
+#### DRF:
+The queue with the lowest share of dominant resource will have higher priority.
+
+#### Proportion:
+The queue with the lowest share will have higher priority.
+
+### 2. JobOrderFn:
+#### Priority:
+Compares jobPriority set in Spec(using PriorityClass) and returns the decision of comparison between two priorities.
+
+#### DRF:
+The job having the lowest share will have higher priority.
+
+#### Gang:
+The job which is not yet ready(i.e. minAvailable number of task is not yet in Bound, Binding, Running, Allocated, Succeeded, Pipelined state) will have high priority.
+
+### 3. JobEnqueueable:
+#### OverCommit:
+After the resource is overCommit, the minResource of the current job plus the inqueue resource of other jobs are lessEqual than the idle resource, the job status is allowed to change from `Pending` to  `Inqueue`, otherwise it will be rejected.
+
+#### Proportion:
+The minResource of the current job plus the allocated & inqueue resources of current queue and minus the elastic resources of current queue are lessEqual than the actual capacity of current queue, the job status is allowed to change from `Pending` to  `Inqueue`, otherwise it will be rejected.
+
+#### ResourceQuota:
+Each resource type of the minResource of the current job lessEqual the resourceQuota of the namespace, the job status is allowed to change from `Pending` to  `Inqueue`, otherwise it will be rejected.
+
+#### SLA:
+Job pending state waiting timeout, the job status is allowed to change from `Pending` to  `Inqueue`, otherwise it will be rejected.
```

Comment:
Actually, `SLA` plugin would not `REJECT` job if its `sla-waiting-time` was not reached, it would just abstain and leave the decision to other plugins such as `overcommit`:

https://github.com/volcano-sh/volcano/blob/ac21d080f0478d03b76def1d7f1af9b4e9538565/pkg/scheduler/framework/session_plugins.go#L393-L419

Reply:
I think his description is only the behavior of sla plugin, For the overall control of jobEnqueueable, I added a description as a whole.

Reply:
Yes, I mean that `SLA` plugin behavior is different with other plugins, which would never reject podgroup from `pending` to `inqueue`. So the description `otherwise it will be rejected` should be modified.

Reply:
Yes, It will be abstained, thks.